python/ribasim/ribasim/model.py

```diff
@@ -638,36 +639,41 @@ def to_xugrid(self, add_flow: bool = False, add_allocation: bool = False):
         uds = uds.assign_coords(from_node_id=(link_dim, from_node_id))
         uds = uds.assign_coords(to_node_id=(link_dim, to_node_id))
 
-        if add_flow:
+        if flow:
             uds = self._add_flow(uds, node_lookup)
-        elif add_allocation:
+        elif allocation:
             uds = self._add_allocation(uds)
 
         return uds
 
-    def _checked_toml_path(self) -> Path:
+    def _toml_path(self) -> FilePath:
+        """Get the path to the TOML file, raising an error if it doesn't exist."""
         toml_path = self.filepath
         if toml_path is None:
-            raise FileNotFoundError("Model must be written to disk to add results.")
-        return toml_path
+            raise FileNotFoundError("Model must be written to disk.")
+        return FilePath(toml_path)
+
+    def _results_dir(self) -> DirectoryPath:
```

Comment:
This could be a public method with @property? And another property with `has_run` or `has_results`?

Reply:
Can do!

Reply:
We now have `model.toml_path` and `model.results_path`, which both error if they don't exist. `model.results_dir` returns the `results_dir` config entry, which is "results" by default.

I didn't add `has_run` or `has_results` yet as I don't think there's much of a need for it yet, but we could always consider it in another PR?